speed up hardswish and hardsigmoid tests

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#38256 speed up hardswish and hardsigmoid tests**

Summary:

Removes hypothesis to speed these tests up, as these tests were flagged as top slow
tests in CI. At the same time, combines the fbgemm and qnnpack test
cases together for better reuse.

Test Plan:

```
python test/test_quantization.py TestQuantizedOps.test_hardswish
python test/test_quantization.py TestQuantizedOps.test_qhardsigmoid
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21506831](https://our.internmc.facebook.com/intern/diff/D21506831)

vkuzo committed:
speed up hardswish and hardsigmoid tests

Summary:

Removes hypothesis to speed these tests up, as these tests were flagged as top slow
tests in CI. At the same time, combines the fbgemm and qnnpack test
cases together for better reuse.

Test Plan:

```
python test/test_quantization.py TestQuantizedOps.test_hardswish
python test/test_quantization.py TestQuantizedOps.test_qhardsigmoid
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar commented on test/quantization/test_quantized_op.py:
```diff
@@ -320,12 +267,46 @@ def test_qsigmoid(self, X):
+        max_sides = (3, 5)
+        side_lens = (1, 7, 8)
+        torch_types = (torch.quint8, torch.qint8)
+        engines = ('fbgemm', 'qnnpack')
```
Maybe use `@override_qengines` instead?

supriyar commented on test/quantization/test_quantized_op.py:
```diff
@@ -320,12 +267,46 @@ def test_qsigmoid(self, X):
+
+                qY = torch.nn.quantized.functional.hardsigmoid(qX)
+                self.assertEqual(qY, qY_hat,
+                                 message="Hardsigmoid failed: {} vs. {}".format(qY, qY_hat))
```
Might help for debugging if we specify which backend this failed on.

malfet commented:
Change looks good to me, but outside of CI environment test already finish in a reasonable amount of time:
```
$ python test/test_quantization.py TestQuantizedOps.test_hardswish TestQuantizedOps.test_qhardsigmoid -v
test_hardswish (quantization.test_quantized_op.TestQuantizedOps) ... /home/nshulga/git/pytorch/test/quantization/test_quantized_op.py:101: UserWarning: 0quantize_tensor_per_tensor_affine current rounding mode is not set to round-to-nearest-ties-to-even (FE_TONEAREST). This will cause accuracy issues in quantized models. (Triggered internally at  /pytorch/aten/src/ATen/native/quantized/affine_quantizer.cpp:25.)
  dtype=torch_type)
ok
test_qhardsigmoid (quantization.test_quantized_op.TestQuantizedOps) ... ok

----------------------------------------------------------------------
Ran 2 tests in 0.619s

OK
```

malfet commented:
I've tried selecting different profiles, as defined in https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_utils.py#L659 but still can not get to 120 sec times reported during CircleCI runs

vkuzo committed:
Update on "speed up hardswish and hardsigmoid tests"


Summary:

Removes hypothesis to speed these tests up, as these tests were flagged as top slow
tests in CI. At the same time, combines the fbgemm and qnnpack test
cases together for better reuse.

Test Plan:

```
python test/test_quantization.py TestQuantizedOps.test_hardswish
python test/test_quantization.py TestQuantizedOps.test_qhardsigmoid
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21506831](https://our.internmc.facebook.com/intern/diff/D21506831)

[ghstack-poisoned]

supriyar commented on test/quantization/test_quantized_op.py:
```diff
@@ -320,12 +267,44 @@ def test_qsigmoid(self, X):
+            if torch.backends.quantized.engine == 'qnnpack' and torch_type != torch.quint8:
+                continue
+
+            with override_quantized_engine(engine):
```
This can be removed as well. The override will take care of it at the test level.

malfet commented:
I know why tests can be slow when run by CI, but fast during single runs:
`hu.array_shapes(1, 8, 1, 8)` returns randomized tensor shape which can be as large as 16 million elements (i.e. `8**8`)
During individual test runs those values are always the same(since we fix the seed), and by chance quite small.
But since the seed is not reset between the runs it can generate a very different (and apparently quite large) tensor shapes when run in sequence with other tests.

malfet commented:
@vkuzo you might need to rebase your change against latest master as I've landed a PR that adds limit to max number of element for hypothesis generated tensors (and it touches the same files are your diff)

vkuzo commented:
> I know why tests can be slow when run by CI, but fast during single runs:
> `hu.array_shapes(1, 8, 1, 8)` returns randomized tensor shape which can be as large as 16 million elements (i.e. `8**8`)
> During individual test runs those values are always the same(since we fix the seed), and by chance quite small.
> But since the seed is not reset between the runs it can generate a very different (and apparently quite large) tensor shapes when run in sequence with other tests.

thanks for the find, makes sense!

vkuzo committed:
Update on "speed up hardswish and hardsigmoid tests"


Summary:

Removes hypothesis to speed these tests up, as these tests were flagged as top slow
tests in CI. At the same time, combines the fbgemm and qnnpack test
cases together for better reuse.

Test Plan:

```
python test/test_quantization.py TestQuantizedOps.test_hardswish
python test/test_quantization.py TestQuantizedOps.test_qhardsigmoid
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21506831](https://our.internmc.facebook.com/intern/diff/D21506831)

[ghstack-poisoned]
